staging/mausezahn.c

```diff
@@ -710,11 +719,20 @@ int getopts (int argc, char *argv[])
 				(ipv6_mode && get_ip6_range_dst(tx.ip_dst_txt, l)) || // returns 1 when no range has been specified
 				(!ipv6_mode && get_ip_range_dst(tx.ip_dst_txt)))
 			{
-				// name2addr4 accepts a DOTTED DECIMAL ADDRESS or a FQDN:
-				if (ipv6_mode)
+				// name2addr{4, 6} accept a valid IP address or a FQDN:
+				if (ipv6_mode) {
 					tx.ip6_dst = libnet_name2addr6 (l, tx.ip_dst_txt, LIBNET_RESOLVE);
-				else
-					tx.ip_dst = libnet_name2addr4 (l, tx.ip_dst_txt, LIBNET_RESOLVE);
+					if (strncmp((char*)&tx.ip6_dst,(char*)&in6addr_error,sizeof(in6addr_error))==0) {
+					  fprintf(stderr, "Failed to set destination IPv6 address. Please check if destination is set to a valid IPv6 address.\n");
+					  return 1;
+					}
+				}
+				else {
+					if ((tx.ip_dst = libnet_name2addr4 (l, tx.ip_dst_txt, LIBNET_RESOLVE)) == -1){
+					  fprintf(stderr, "Failed to set destination IPv4 address. Please check if destination is set to a valid IPv4 address.\n");
+					  return 1;
+					}
+			  }
 			}
```

Comment:
So in the case of invalid IPv4 address we get -1 which is 255.255.255.255 IPv4 address which is OK to test for invalid host resolving, but not for IPv4 address itself I think, as 255.255.255.255 is a still valid addres to use it in the IPv4 header. The similar for IPv6, so ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff might be a valid IPv6 address. I think that it is better to check for valid IPv4/6 address format with inet_pton(...) function.